hooks.go

```diff
@@ -0,0 +1,113 @@
+package easypost
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"io"
+	"net/url"
+	"time"
+)
+
+// HookEvent is the base type for all hook events
+type HookEvent struct {
+}
+
+// HookEventSubscriber is the base type for all hook event subscribers.
+// ID needs to be unique in order to find and remove the subscriber
+type HookEventSubscriber struct {
+	ID string // need unique ID for removal
+}
```

Comment:
Question: why do we need these two struct?

Reply:
For shared inheritance of the `Request` and `Response` events/subscribers